src/components/VerticalNavigation/VerticalNavigation.js

```diff
@@ -0,0 +1,313 @@
+  // - The depth prop will automatically be set to the currently nested depth (primary, secondary or tertiary).
+  // - Any props passed explicitly to VerticalNavigationItem children will take precendence over inherited props.
+  static renderChildren(props) {
+    const { children, ...parentProps } = props; // TODO maybe take specific props instead of passing all of them?
+    const { primaryItem, secondaryItem, tertiaryItem } = props;
+    const childDepth = VerticalNavigation.getNextDepth(props.depth);
+    return React.Children.map(children, child => {
+      if (child.type === VerticalNavigationItem) {
+        const item = VerticalNavigation.getItemObject(child.props);
+        const primary = childDepth === 'primary' ? item : primaryItem;
+        const secondary = childDepth === 'secondary' ? item : secondaryItem;
+        const tertiary = childDepth === 'tertiary' ? item : tertiaryItem;
+        const childProps = {
+          ...parentProps,
+          depth: childDepth,
+          primaryItem: primary,
+          secondaryItem: secondary,
+          tertiaryItem: tertiary,
+          ...child.props, // We include this to allow overriding these props inline on a VerticalNavigationItem.
+        };
+        return <VerticalNavigationItem {...childProps} />;
+      } else {
+        return child;
+      }
+    });
+  }
+
+  static handleItemClick(primary, secondary, tertiary, inMobileState) {
+    const item =
+      (primary && secondary && tertiary) || (primary && secondary) || primary;
+    if (inMobileState) {
+      if (item.children && item.children.length > 0) {
+        this.updateMobileMenu(primary, secondary, tertiary); // TODO figure out what this did in ng
+      } else {
+        this.updateMobileMenu(); // TODO figure out what this did in ng (expanded states?)
+      }
+    }
+    if (!inMobileState || !item.children || item.children.length === 0) {
+      VerticalNavigation.navigateToItem(item);
+    }
+  }
+
+  static getNextDepth(depth) {
+    if (!depth) return 'primary';
+    return depth === 'primary' ? 'secondary' : 'tertiary';
+  }
+
+  static getItemObject(props) {
+    return {
+      title: props.title,
+      trackActiveState: props.trackActiveState,
+      trackHoverState: props.trackHoverState,
+      mobileItem: props.mobileItem,
+      iconStyleClass: props.iconStyleClass,
+      badges: props.badges,
+      children:
+        props.children &&
+        props.children.length > 0 &&
+        React.Children.map(props.children, child =>
+          VerticalNavigation.getItemObject(child.props),
+        ),
+    };
+  }
+
+  static navigateToItem(item) {
+    console.log('TODO NAVIGATE!', item); // TODO
+  }
+
+  constructor() {
+    super();
+    this.state = {
+      hoverSecondaryNav: false,
+      hoverTertiaryNav: false,
+      collapsedSecondaryNav: false,
+      collapsedTertiaryNav: false,
+    };
+    this.handleNavBarToggleClick = this.handleNavBarToggleClick.bind(this);
```

Comment:
We can use the bind helpers in helpers.js here now:
https://github.com/patternfly/patternfly-react/blob/master/src/common/helpers.js

Reply:
(see Wizard examples)